From 9ae9a07e7572fcfc777523447e202d50e7b91896 Mon Sep 17 00:00:00 2001
From: Minh Le <minh.le.aj@renesas.com>
Date: Thu, 8 May 2025 17:14:05 +0700
Subject: [PATCH 19/46] i2c: rcar: Use fixed clock value for R-Car Gen5

Use fixed clock value for R-Car Gen5 I2C,
with SCPBUSD4_SCP_MAIN at 150Mhz forchannel 0,
SGASYNCD8_PERW_MAIN at 133.33Mhz for remaining channels (1-8)

Signed-off-by: Minh Le <minh.le.aj@renesas.com>
---
 drivers/i2c/busses/i2c-rcar.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/i2c/busses/i2c-rcar.c b/drivers/i2c/busses/i2c-rcar.c
index 1a293c772492..95658eec6558 100644
--- a/drivers/i2c/busses/i2c-rcar.c
+++ b/drivers/i2c/busses/i2c-rcar.c
@@ -121,6 +121,8 @@
 #define ID_P_PM_BLOCKED		BIT(31)
 #define ID_P_MASK		GENMASK(31, 28)
 
+#define CFG_CLK_IGNORE
+
 /* VDK-only: Apply DMA workarounds for R-Car X5H */
 #define VDK	1
 
@@ -338,7 +340,11 @@ static int rcar_i2c_clock_calculate(struct rcar_i2c_priv *priv)
 	 * SCL	= clkp / (8 + SMD * 2 + SCLD + SCHD +F[(ticf + tr + intd) * clkp])
 	 *
 	 */
+#if !defined(CFG_CLK_IGNORE)
 	rate = clk_get_rate(priv->clk);
+#else
+	rate = (priv->io == (void __iomem *)0xc11d0000) ? 150000000 : 133333333;
+#endif
 
 	if (!priv->fast_mode_plus) {
 		cdf = rate / 20000000;
@@ -1145,11 +1151,13 @@ static int rcar_i2c_probe(struct platform_device *pdev)
 	if (!priv)
 		return -ENOMEM;
 
+#if !defined(CFG_CLK_IGNORE)
 	priv->clk = devm_clk_get(dev, NULL);
 	if (IS_ERR(priv->clk)) {
 		dev_err(dev, "cannot get clock\n");
 		return PTR_ERR(priv->clk);
 	}
+#endif
 
 	priv->io = devm_platform_get_and_ioremap_resource(pdev, 0, &priv->res);
 	if (IS_ERR(priv->io))
-- 
2.34.1

